### ISSUE TITLE
Dependency issue with the gradle plugin

### ISSUE THREAD

**Issue by jmini** [OpenAPITools/openapi-style-validator#288]
**Describe the bug**

When using the `openapi-style-validator` version 1.7, there is following error:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'exampleUsage'.
> Could not resolve all files for configuration ':classpath'.
   > Could not find org.openapitools.openapistylevalidator:gradle-plugin:1.7.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/openapitools/openapistylevalidator/gradle-plugin/1.7/gradle-plugin-1.7.pom
       - https://plugins.gradle.org/m2/org/openapitools/openapistylevalidator/gradle-plugin/1.7/gradle-plugin-1.7.pom
     Required by:
         project : > org.openapitools.openapistylevalidator:org.openapitools.openapistylevalidator.gradle.plugin:1.7

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s
```

**To Reproduce**
Setup:

File `build.gradle`
```gradle
plugins {
  id 'org.openapitools.openapistylevalidator' version '1.7'
}

openAPIStyleValidator {
    // set the input file option:
    inputFile = file('openapi.yaml')

    // customize the validation options:
    validateInfoLicense = false
    validateInfoDescription = false
    validateInfoContact = false
    validateOperationDescription = false
    validateOperationTag = false
    validateOperationSummary = false
}
```

File `settings.gradle`
```gradle
pluginManagement {
    repositories {
        mavenCentral()
        gradlePluginPortal()
    }
}

rootProject.name = 'exampleUsage'

```

File `openapi.yaml`
```yaml
openapi: 3.0.1
info:
  title: ping test
  version: '1.0'
servers:
  - url: 'http://localhost:9999/'
paths:
  /ping:
    post:
      operationId: pingGet
      responses:
        '201':
          description: OK
```

Run `gradle openAPIStyleValidator`

**Expected behavior**
The plugin works.

**Additional context**
It is working with version `1.4` of the `org.openapitools.openapistylevalidator` plugin.
Broken with all versions since then.

Tested with (`gradle --version`):
```
------------------------------------------------------------
Gradle 7.3
------------------------------------------------------------

Build time:   2021-11-09 20:40:36 UTC
Revision:     96754b8c44399658178a768ac764d727c2addb37

Kotlin:       1.5.31
Groovy:       3.0.9
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          1.8.0_202 (Azul Systems, Inc. 25.202-b05)
OS:           Mac OS X 10.15.2 x86_64
```

**Comment by jmini** [comment]
*Workaround:*

At the top of your `build.gradle` file, add the correct dependency explicitly:

```
buildscript {
    repositories {
        mavenCentral()
    }
    dependencies {
        classpath 'org.openapitools.openapistylevalidator:openapi-style-validator-gradle-plugin:1.7'
    }
}
```

---

With this: `gradle openAPIStyleValidator` produces:
```

> Task :openAPIStyleValidator
Validating spec: /Users/jbr/tmp/gradletmp10/openapi.yaml
Validating with options: ValidatorParameters [validateInfoLicense=false, validateInfoDescription=false, validateInfoContact=false, validateOperationOperationId=true, validateOperationDescription=false, validateOperationTag=false, validateOperationSummary=false, validateModelPropertiesExample=true, validateModelPropertiesDescription=true, validateModelRequiredProperties=true, validateModelNoLocalDef=true, validateNaming=true, ignoreHeaderXNaming=true, pathNamingConvention=HyphenCase, headerNamingConvention=UnderscoreUpperCase, parameterNamingConvention=CamelCase, propertyNamingConvention=CamelCase]

BUILD SUCCESSFUL in 7s
1 actionable task: 1 executed
```

**Comment by jmini** [comment]
Analysis of the issue:

In our `build.gradle` of the gradle plugin we are changing the artifact id of the published jar:

https://github.com/OpenAPITools/openapi-style-validator/blob/1eae52a6883dbc4a59415cea0b168e2931d9330c/gradle-plugin/build.gradle#L81

But our generated [marker plugin artifact](https://docs.gradle.org/current/userguide/plugins.html#sec:plugin_markers) is not aware of the change.
If you look at the [POM file](https://repo.maven.apache.org/maven2/org/openapitools/openapistylevalidator/org.openapitools.openapistylevalidator.gradle.plugin/1.7/org.openapitools.openapistylevalidator.gradle.plugin-1.7.pom) of the Marker Artifact with coordinates:
```xml
<groupId>org.openapitools.openapistylevalidator</groupId>
<artifactId>org.openapitools.openapistylevalidator.gradle.plugin</artifactId>
<version>1.7</version>
```

You will see the wrong dependency:
```xml
 <dependencies>
    <dependency>
        <groupId>org.openapitools.openapistylevalidator</groupId>
        <artifactId>gradle-plugin</artifactId>
        <version>1.7</version>
    </dependency>
</dependencies>
```

It should be:
```xml
<dependencies>
    <dependency>
        <groupId>org.openapitools.openapistylevalidator</groupId>
        <artifactId>openapi-style-validator-gradle-plugin</artifactId>
        <version>1.7</version>
    </dependency>
</dependencies>
```

---
Analysis of the different marker POM files

* [1.3](https://repo.maven.apache.org/maven2/org/openapitools/openapistylevalidator/org.openapitools.openapistylevalidator.gradle.plugin/1.3/org.openapitools.openapistylevalidator.gradle.plugin-1.3.pom) ✅ 
* [1.4](https://repo.maven.apache.org/maven2/org/openapitools/openapistylevalidator/org.openapitools.openapistylevalidator.gradle.plugin/1.4/org.openapitools.openapistylevalidator.gradle.plugin-1.4.pom) ✅ 
* [1.5](https://repo.maven.apache.org/maven2/org/openapitools/openapistylevalidator/org.openapitools.openapistylevalidator.gradle.plugin/1.5/org.openapitools.openapistylevalidator.gradle.plugin-1.5.pom) ❌
* [1.6](https://repo.maven.apache.org/maven2/org/openapitools/openapistylevalidator/org.openapitools.openapistylevalidator.gradle.plugin/1.6/org.openapitools.openapistylevalidator.gradle.plugin-1.6.pom) ❌
* [1.7](https://repo.maven.apache.org/maven2/org/openapitools/openapistylevalidator/org.openapitools.openapistylevalidator.gradle.plugin/1.7/org.openapitools.openapistylevalidator.gradle.plugin-1.7.pom) ❌

**Comment by jmini** [comment]
* `1.4` was published with gradle `6.5.1`
* `1.5`, `1.6` and `1.7` were published with gradle `6.8`

We need to check if the issue is solved with more recent versions of Gradle.
